### ISSUE TITLE
load profile from json

### ISSUE THREAD

**Issue by parisni** [davidhay25/clinFHIR#17]
Hi

I guess it would be usefull to be able to load profile from json, instead of building them from scrach by clicking on buttons.
The scenario builder allows to import resources from json. What about provinding this feature for tue logical Modeler ?

Is there a workaround ?

Thanks

**Comment by davidhay25** [comment]
I presume you mean an existing logical model created in some other tool or manually?

**Comment by parisni** [comment]
Yeah, exactly, an other tooling for easily creating profiles (I guess profile = logical model nope ?)

**Comment by davidhay25** [comment]
Ah no - which is why I checked! The logical model uses the FHIR
infrastructure (including datatypes, StructureDefinition resource etc) but
is not the same as a profile. For example, there is no need for extensions,
you just add another element...

In theory you could post in a profile and convert it to a logical model by
resolving the extensions, (which is what the scenario builder does) but it
will never be able to handle all the complexity of profiles (I only support
a limited subset of capability - it's complicated under the hood!)

Sorry about that - but, of course, Forge (https://fhir.furore.com/forge/)
is able to load and edit profiles, so that will be your best bet moving
forward...

cheers...

**Comment by parisni** [comment]
David

This clarifies.

I tried the last forge version (STU3). Apparently, it does not support to import profiles as json but only as xml.

Or I am missing something ?

**Comment by davidhay25** [comment]
No I think that it does mostly work with XML. But assuming that you have a
valid profile (StructureDefinition) it's easy to convert from json to XML
either by saving to a FHIR server in json & retrieving it in XML - or the
command line tools provided by the spec (note that I'm not familiar in
detail with these - but they do exist).

cheers...
